packages/react/src/components/AccountSettings/SetupTOTP/SetupTOTP.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import QRCode from 'qrcode';
+
+import {
+  AmplifyUser,
+  getLogger,
+  getTotpCodeURL,
+  setupTOTP,
+  translate,
+  verifyTOTPToken,
+} from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { View, Flex } from '../../../primitives';
+import { FormValues } from '../types';
+import { ComponentClassName } from '../constants';
+import {
+  ConfirmationCode,
+  CopySecretKey,
+  Error,
+  SecretKeyQRCode,
+  SubmitButton,
+} from './defaults';
+import { ConfigureTOTPProps, TotpSecret, VerifyTotpStatus } from './types';
+import { QR_CODE_DIMENSIONS } from './constants';
+
+const logger = getLogger('Auth');
+
+function SetupTOTP({
+  totpIssuer = 'AWSCognito',
+  totpUsername,
+  onSuccess,
+  onError,
+}: ConfigureTOTPProps): JSX.Element | null {
+  const [formValues, setFormValues] = React.useState<FormValues>({ code: '' });
```

Comment:
Unless we plan to add more keys to formValues in the immediate future that this can be simplified to:
```
const [confirmationCode, setConfirmationCode] = React.useState('');
```

Reply:
Done in 7d340ca6bcb105e8ab9de9b6870d7b51891300b3